### PR TITLE
Split library_gl.js into library_webgl.js and library_webgl2.js.

### PR DESCRIPTION
**Pull request by juj**
Move all WebGL 2 and FULL_ES3 specific symbols to their own file. Also refactors the proc address retrieval to be built as separate libs for GL emu, WebGL 1 and WebGL 2, to avoid depending on symbols that might not exist in the build.

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -2707,8 +2707,9 @@ def path_to_system_js_libraries(library_name):
       'c': '',
       'dl': '',
       'EGL': 'library_egl.js',
-      'GL': 'library_gl.js',
-      'GLESv2': 'library_gl.js',
+      'GL': 'library_webgl.js',
+      'GLESv2': 'library_webgl.js',
+      'GLESv3': ['library_webgl.js', 'library_webgl2.js'],
```
Comment (hujiajie) [inline tools/shared.py]: Maybe I've missed something, but I think OpenGL ES 3.0+ are traditionally provided by libGLESv2.so instead of libGLESv3.so?

Comment (juj) [reply tools/shared.py]: Oh yeah, that's right. Updated.

**Comment by juj** [comment]
This now receives contents of #7671 to also resolve the issue that GetProcAddress with WebGL 1 would also include WebGL 2 symbols, even if one did not build with WebGL 2 enabled.

File: embuilder.py [hunk 2]
```diff
@@ -246,6 +246,13 @@ def main():
           return int(emscripten_GetProcAddress("waka waka"));
         }
       ''', ['libgl-mt.bc'], ['-s', 'USE_PTHREADS=1'])
+    elif what == 'gl-webgl2':
+      build('''
+        extern "C" { extern void* emscripten_GetProcAddress(const char *x); }
+        int main() {
+          return int(emscripten_GetProcAddress("waka waka"));
+        }
+      ''', ['libgl-webgl2.bc'], ['-s', 'USE_WEBGL2=1'])
```
Comment (kripken) [inline embuilder.py]: all additions to embuilder should be added to `test_embuilder`, and to either `USER_TASKS` or `SYSTEM_TASKS` in the embuilder file.

File: system/lib/gl/gl.c [hunk 3]
```diff
@@ -1575,999 +1770,33 @@ void* emscripten_GetProcAddress(const char *name_) {
...
-  if (!strcmp(name, "glFramebufferTexture3D")) return emscripten_glFramebufferTexture3D;
-  if (!strcmp(name, "glTexImage2D")) return emscripten_glTexImage2D;
-  if (!strcmp(name, "glGetShaderPrecisionFormat")) return emscripten_glGetShaderPrecisionFormat;
-
-  // If LEGACY_GL_EMULATION is on, allow access to those functions too
-#ifdef LEGACY_GL_EMULATION
-  if (!strcmp(name, "glDeleteObject")) return emscripten_glDeleteObjectARB;
-  if (!strcmp(name, "glGetObjectParameteriv")) return emscripten_glGetObjectParameterivARB;
-  if (!strcmp(name, "glGetInfoLog")) return emscripten_glGetInfoLogARB;
-  if (!strcmp(name,
```
Comment (kripken) [inline system/lib/gl/gl.c]: why are these methods not in `emscripten_legacy_gl_emulation_GetProcAddress`?

Comment (juj) [reply system/lib/gl/gl.c]: Yeah, can move them there.

File: embuilder.py (hunk 2 above)
Comment (juj) [reply embuilder.py]: What is the role of that test? The CI `build` step already does `python embuilder.py build ALL`, which builds all the variants of `gl`, and verifies that the individual generated .bc files exist?

Comment (juj) [reply embuilder.py]: Added the different variants in the build

Comment (kripken) [reply embuilder.py]: The test verifies the specific files that are generated, and guards against "ALL" not being complete. But yeah, it's not very clever or in-depth or anything like that.

**Comment by kripken** [comment]
@juj this caused a regression on an important application. Trying to investigate, it looks like the relevant facts are

1. The application is built with pthreads, but the relevant code here is single-threaded.
2. Before this landed, we did not link in the compiled C code for `glBufferData` that does the proxying. After this PR, we do link it in. (Is this intended?)
3. That ends up being a problem, since that proxying code checks `currentThreadOwnsItsWebGLContext`. But `InitWebGLTls` has not been called, since `emscripten_webgl_create_context` has not been called (the user code does not call it, and I don't think it's necessary for code to do so?).

**Comment by juj** [comment]
>     2\. Before this landed, we did not link in the compiled C code for `glBufferData` that does the proxying. After this PR, we do link it in. (Is this intended?)

The WebGL proxying code should be included exactly only when both `-s OFFSCREEN_FRAMEBUFFER=1` and `-s USE_PTHREADS=1` are enabled, since that is what enables/activates the proxying.

>     3\. But `InitWebGLTls` has not been called, since `emscripten_webgl_create_context` has not been called

>     3\. it ends up calling `emscripten_async_queue_on_thread`

What function ends up calling `emscripten_async_queue_on_thread`? `glBufferData`? If so, it is not valid to call `glBufferData` (or any other GL function) in a thread without first having called `emscripten_webgl_create_context` + `emscripten_webgl_make_current` functions to have an active context in the calling thread.

**Comment by kripken** [comment]
I see, thanks @juj! Ok, then what I think is going wrong is that `OFFSCREEN_FRAMEBUFFER` is off, but that code comes in. I'll look into a PR.

(yes, it's glBufferData that ends up calling the async proxy code)

**Pull request merged**